htdocs/banners.php

```
@@ -98,20 +99,23 @@ function clientlogin()
-    if ($_SESSION['banner_login'] == '' || $_SESSION['banner_pass'] == '') {
+    $banner_login = XoopsRequest::getString('banner_login', '', 'SESSION');
+    $banner_pass  = XoopsRequest::getString('banner_pass', '', 'SESSION');
+    if ($banner_login == '' || $banner_pass == '') {
         redirect_header('banners.php', 2, _BANNERS_NO_LOGIN_DATA);
     }
-    $sql = sprintf('SELECT cid, name, passwd FROM %s WHERE login=%s', $xoopsDB->prefix('bannerclient'), $xoopsDB->quote($_SESSION['banner_login']));
+    $sql    = sprintf('SELECT cid, name, passwd FROM %s WHERE login=%s', $xoopsDB->prefix('bannerclient'), $xoopsDB->quote($banner_login));
     $result = $xoopsDB->query($sql);
-    if (!$xoopsDB->isResultSet($result)) {
+    if (!$xoopsDB->isResultSet($result) || !$result instanceof \mysqli_result) {
         throw new \RuntimeException(
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '112', column '19'). (undefined)

(MissingImport)